src/main.rs

```diff
@@ -385,6 +387,15 @@ fn collect_files(
         let in_file = if using_stdin {
             InFile::StdIn
         } else {
+            //skip non png files if recusive flag is used.
+            //(still allow the user to convert a non png file if recusive flag is not used);
+            if !top_level && recursive && {
```

Comment:
We don't need to check the recursive flag here, the top_level check is sufficient
```suggestion
            // Skip non png files if not given on top level
            if !top_level && {
```